flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisConsumer.java

```diff
@@ -242,6 +244,8 @@ public FlinkKinesisConsumer(
                     "Flink Kinesis Consumer is going to read the following streams: {}",
                     sb.toString());
         }
+
+        this.shardAssigner = createShardAssigner();
```

Comment:
While this works, I have a couple of recommendations I would like to hear your thoughts on:
1. We should move this logic into the Table API wrapper. Unless you see benefit in this being configurable via properties for the DS connector. I see value in them being consistent.
2. Since you are using an enum this implementation violates the `O` in `SOLID` since it is closed for extension. Users can already set custom shard assigners in the datastream API but this will not allow custom assigners in Table API. What are you thoughts on using service loader to allow shard assigners to be loaded dynamically? Similar to how table connectors are loaded today

Reply:
+1 for dynamically resolving additional shard assigners using service loader when using Table API.

Reply:
I think that's a great suggestion. Are there any other aspects of this connector (or any others) that are extensible in this way? Other than the connectors themselves? Hoping to emulate if such a thing already exists... Thanks

Reply:
I've made the requested changes in the latest (amended) commit. Would appreciate feedback/suggestions. Thanks

Reply:
> @karthitect please fix commit message as per the check. The PR title looks good: `[FLINK 30777][Connectors/Kinesis] Allow Kinesis Table API Connector to specify shard assigner`

Done

Reply:
> Your build is failing due to spotbugs errors. Please run `mvn clean package` and verify the full build before pushing PR

Done

Reply:
> PR title missing `-` in issue id, triggering Title Validator failure. Should be `[FLINK-30777] [Connectors/Kinesis] Allow Kinesis Table API Connector to specify shard assigner`

Done